component/registry.go

```diff
@@ -61,6 +62,14 @@ type Options struct {
 	// attribute denoting the component ID.
 	Tracer trace.TracerProvider
 
+	// Clusterer allows components to work in a clustered fashion. The
+	// clusterer is shared between all components initialized by a Flow
+	// controller.
+	//
+	// TODO(@tpaschalis) As noted in the ComponentGlobals, we most likely only
+	// need the Sharder implementation here.
+	Clusterer *cluster.Clusterer
```

Comment:
I think this potentially fits in with finding a better way to send signals to components through the Update method so we don't have to keep adding new fields to the options. 

But: not a blocker for this PR.